src/codegen/codegen_c_visitor.cpp

```diff
@@ -729,7 +729,7 @@ bool CodegenCVisitor::is_constant_variable(std::string name) {
     auto symbol = program_symtab->lookup_in_scope(name);
     bool is_constant = false;
     if (symbol != nullptr) {
-        if (symbol->has_any_property(NmodlType::read_ion_var)) {
+        if (symbol->has_any_property(NmodlType::read_ion_var) && optimize_ion_variable_copies()) {
```

Comment:
@sharkovsky : `read_ion_var` here is an internal copy of the mechanism. So this can't be const anyway because we have to make copy. So I would just remove this if block.

Reply:
Actually i discussed this with @sharkovsky and he has a point. In his case he did explicitely want `optimize_ion_variable_copies` to be `false` and so the check makes sense. The rationale is that in some derived classes this could be set to `false` too.

Reply:
@pramodk if I understand correctly, you're saying that _regardless_ of whether `optimize_ion_variable_copies` returns true or false, we should not mark the symbol constant when `symbol->has_any_property(NmodlType::read_ion_var)` evaluates to true. 
So you're saying the we should remove the `if` statement altogether?

Question: if we have to make copies anyway, why did the error arise only when I set `optimize_ion_variable_copies` to false?

Reply:
> Question: if we have to make copies anyway, why did the error arise only when I set optimize_ion_variable_copies to false?

as discussed offline, when `optimize_ion_variable_copies` is true, those variables are not used and hence error is not generated. Making variables const was bug in the first place.